internal/cli/atlas/deployments/search/indexes/list.go

```diff
@@ -0,0 +1,169 @@
+	"github.com/mongodb/mongodb-atlas-cli/internal/mongodbclient"
+	"github.com/mongodb/mongodb-atlas-cli/internal/podman"
+	"github.com/mongodb/mongodb-atlas-cli/internal/store"
+	"github.com/mongodb/mongodb-atlas-cli/internal/usage"
+	"github.com/spf13/cobra"
+	atlasv2 "go.mongodb.org/atlas-sdk/v20230201007/admin"
+)
+
+var listTemplate = `ID	NAME	DATABASE	COLLECTION{{range .}}
+{{.ID}}	{{.Name}}	{{.Database}}	{{.Collection}}{{end}}
+`
+
+type ListOpts struct {
+	cli.GlobalOpts
+	cli.OutputOpts
+	options.DeploymentOpts
+	search.IndexOpts
+	mongodbClient mongodbclient.MongoDBClient
+	store         store.SearchIndexLister
+}
+
+func (opts *ListOpts) Run(ctx context.Context) error {
+	if err := opts.validateAndPrompt(ctx); err != nil {
+		return err
+	}
+
+	err := opts.RunLocal(ctx)
+	if err != nil && (errors.Is(err, options.ErrDeploymentNotFound)) {
+		return opts.RunAtlas()
+	}
+
+	return err
+}
+
+func (opts *ListOpts) RunAtlas() error {
+	r, err := opts.store.SearchIndexes(opts.ConfigProjectID(), opts.DeploymentName, opts.DBName, opts.Collection)
+	if err != nil {
+		return err
+	}
+
+	return opts.Print(newSearchIndexDefinition(r))
+}
+
+func (opts *ListOpts) RunLocal(ctx context.Context) error {
+	connectionString, err := opts.ConnectionString(ctx)
+	if err != nil {
+		return err
+	}
+
+	if err = opts.mongodbClient.Connect(ctx, connectionString, connectWaitSeconds); err != nil {
+		return err
+	}
+	defer opts.mongodbClient.Disconnect(ctx)
+
+	r, err := opts.mongodbClient.Database(opts.DBName).SearchIndexes(ctx, opts.Collection)
+	if err != nil {
+		return err
+	}
+
+	return opts.Print(r)
+}
+
+func (opts *ListOpts) initMongoDBClient() error {
+	opts.mongodbClient = mongodbclient.NewClient()
+	return nil
+}
+
+func (opts *ListOpts) initStore(ctx context.Context) func() error {
+	return func() error {
+		var err error
+		opts.store, err = store.New(store.AuthenticatedPreset(config.Default()), store.WithContext(ctx))
+		return err
+	}
+}
+
+func (opts *ListOpts) validateAndPrompt(ctx context.Context) error {
+	if opts.DBName == "" {
+		if err := promptRequiredName("Database", &opts.DBName); err != nil {
+			return err
+		}
+	}
+
+	if opts.Collection == "" {
+		if err := promptRequiredName("Collection", &opts.Collection); err != nil {
+			return err
+		}
+	}
+
+	if opts.DeploymentName == "" {
+		if err := opts.DeploymentOpts.Select(ctx); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func newSearchIndexDefinition(indexes []atlasv2.ClusterSearchIndex) []*mongodbclient.SearchIndexDefinition {
+	out := make([]*mongodbclient.SearchIndexDefinition, len(indexes))
+	for i, v := range indexes {
+		out[i] = &mongodbclient.SearchIndexDefinition{
+			ID:         *v.IndexID,
+			Name:       v.Name,
+			Collection: v.CollectionName,
+			Database:   v.Database,
+		}
+	}
+
+	return out
+}
+
+func ListBuilder() *cobra.Command {
+	opts := &ListOpts{}
+	cmd := &cobra.Command{
+		Use:   "list",
+		Short: "List all Atlas Search indexes for a deployment.",
```

Comment:
```suggestion
		Short: "List all Atlas Search indexes for a deployment.",
		Aliases: []string{"ls"},
```
for consistency with other list commands